src/Platforms/AbstractPlatform.php

```diff
@@ -2887,10 +2887,29 @@ public function getColumnDeclarationSQL($name, array $column)
 
             $notnull = ! empty($column['notnull']) ? ' NOT NULL' : '';
 
-            $unique = ! empty($column['unique']) ?
-                ' ' . $this->getUniqueFieldDeclarationSQL() : '';
+            if (! empty($column['unique'])) {
+                Deprecation::trigger(
+                    'doctrine/dbal',
+                    'https://github.com/doctrine/dbal/pull/5656',
+                    'The usage of the "unique" column property is deprecated. Use unique constraints instead.',
```

Comment:
One the one hand, I feel these messages should include more context as a way to make it more obvious to users receiving them several times that they are making progress, on the other end it seems we only have access to the column name and not to the table name here.

Reply:
I don't think using runtime deprecations for tracking progress in this is an essential use case here. Database schema definitions are mostly static, so it should be enough to go through them once and make sure that this deprecation no longer triggers.

As for making the error messages more user-friendly, there's the same dilemma here as in https://github.com/doctrine/dbal/issues/5645. On the one hand, we want the method to accept as few parameters as necessary while on the other hand we want the error messages to be as clear as possible.

Given that these messages are temporary, the currently proposed implementation looks good enough to me.